Global tokens spacing bump to 0.36

rohanprasadofficial opened the pull request:
### Platforms Impacted
- [x] iOS
- [x] macOS
- [x] win32 (Office)
- [x] windows
- [x] android

### Description of changes

This PR includes following changes: 
-  Update all design-tokens packages in FURN to 0.31.0  -->  0.36.0 
-  Update FURN accordingly to the new spacing to size tokens 
-  Tester updates for Spacing Component 
-  Snapshot updates 
-  Shadow mapping updates (as now shadow is treated as pipelineOutputShadow.shadow2[0] instead of pipelineOutputShadow.global.shadow2[0] 


I also tried to organize my commits - might be easier to review commit by commit inspired by @lyzhan7.  

**Tokens Mapping:** 

globalTokens.spacing.none ->  globalTokens.sizeNone
globalTokens.spacing.xxs ->  globalTokens.size20
globalTokens.spacing.xs ->  globalTokens.size40
globalTokens.spacing.sNudge  ->  globalTokens.size60
globalTokens.spacing.s ->  globalTokens.size80 
globalTokens.spacing.mNudge  ->  globalTokens.size100
globalTokens.spacing.m ->  globalTokens.size120
globalTokens.spacing.l ->  globalTokens.size160
globalTokens.spacing.xl ->  globalTokens.size200
globalTokens.spacing.xxl ->  globalTokens.size240
globalTokens.spacing.xxxl  ->  globalTokens.sizee320

New Extra Added tokens for size: 

 globalTokens.size280
 globalTokens.size360
 globalTokens.size400
 globalTokens.size480
 globalTokens.size560
 globalTokens.size640
 globalTokens.size720
 globalTokens.size800
 globalTokens.size1200 



### **Update :**  

Now every default , win32, macos files mapped to win32/macos ramp up. 
Android and iOS specific files maps to the  android & iOS ramp now.  
Exception : in the files [iOS Button component](https://github.com/microsoft/fluentui-react-native/blob/main/packages/components/Button/src/ButtonTokens.ios.ts#L17) , there is a place using sNudge but there is mapping in the tshirt size naming so i have mapped it to size100 , as i made sense as sNudge lies between s (size120)  and xs (size80) for mobile. 


Note as per our discussion on previous PR here ( https://github.com/microsoft/fluentui-react-native/pull/2354 ) which is abandoned now ( i messed some branches , hence raising a new one  😭)  


 ---------------------------------------------------------------------------------------------------------------------------------------
@rurikoaraki 's comments : 
All values should default to web/desktop values except for android-specific files/mobile components. Ian had a conversion chart, please look at that to convert to the correct values for desktop. 

@lyzhan7 's comments : 

It looks like all the components got updated with the old iOS/Android ramp rather than the old win32/desktop ramp. You can also see the conversion here: https://github.com/microsoft/fluentui-design-tokens/pull/36/files

For corner radius and stroke width we ran into a similar problem where there were two different ramps (ex. on mobile, corner.radius.extraLarge was defined as 120 but on win32 it was 80) so it wasn't obvious what to update a reference to corner.radius.extraLarge to. For corner radius/stroke width we ended up using the desktop mappings for everything, including mobile components. The reasoning was that up until the recent iOS/Android token work, iOS was actually referencing windows tokens, and so most components were built with the assumption that the windows values were being used - using the desktop conversion would cause less unwanted change than the using the mobile conversion.

I would prefer if we stayed consistent here and just updated everything with the desktop conversion, but also fine if we wanted to use the mobile conversion table for ios/android specific files. Eventually I think we should be going through each component to check that with all the tokens changes they still align with the designs.


------------------------------------------------------------------------------------------------------------------------------------------

We have different in Desktop token and Mobile token mapping to the sizes hence , my previous PR was mapped to mostly Android. 
To be consistent across the updation i have updated the all the spacing tokens to size tokens which reference to the Desktop platforms mapping based on this PR.
https://github.com/microsoft/fluentui-design-tokens/pull/36/files

What's was blocking 
On Desktop platforms :  globalTokens.spacing.xs ->  globalTokens.size40
But on Mobile platforms : globalTokens.spacing.xs ->  globalTokens.size80
thus the difference comes in , but as discussed iOS/ Android was actually referencing windows tokens, and so most components were built with the assumption that the windows values were being used - using the desktop conversion would cause less unwanted change than the using the mobile conversion. 

We have to go to each of the Mobile Component Tokens update the spacing token to correct values top reflect the correct values there. 

Also Note :  packages/theming/apple-theme/src/appleTheme.ios.ts have updates in the spacing tokens. As we have discuss to not to include the spacing ramp in the themes , i think that would make more sense to have a separate PR to tackle that but for now , we have update the types mapping , it can be removed as a part of removal of spacing from theme. 






(a summary of the changes made to files)

- <component>Tokens.ts 
   All the mapping updates from spacing to sizes ramp. 
-  Tester App Updates 
   apps/fluent-tester/src/TestComponents/Spacing/SpacingTest.tsx
- Snapshot updates 
-  package.json  and yarn.lock files - pacakage bumps. 



### Verification
Visual could reflect on iOS and Android , but verified on Button , Checkbox - Android - Few things needs updating to this ramp Made sure the tester starts up without issues on iOS and that none of the pages cause unexpected errors
Desktop Platform needs a visual verification once. 

### Pull request checklist

This PR has considered (when applicable):
- [ ] Automated Tests
- [ ] Documentation and examples
- [ ] Keyboard Accessibility
- [ ] Voiceover
- [ ] Internationalization and Right-to-left Layouts

rohanprasadofficial commented:
@gulnazsayed @ValentinaKozlova @Saadnajmi @rurikoaraki , I would really appreciate if you can give quick look the win/mac components once visually once if everything looks not broken 😃  .  As recently we have shipped many components with spacing values. If any issues please let me know :)  


On the android side , @ayush547 @ankraj12 and me, we can verify Checkbox ,Switch ,  Button and Avatar on the verification. 
@lyzhan7  , can you help with iOS , if any components recently used Spacing values. 

Thanks

ankraj12 commented on packages/components/Badge/src/__tests__/__snapshots__/Badge.test.tsx.snap:
```diff
@@ -17,7 +17,7 @@ exports[`Badge component tests Badge all props 1`] = `
       "minHeight": 24,
       "minWidth": 24,
-      "paddingHorizontal": 8,
+      "paddingHorizontal": 4,
```
Are these changes expected? The visuals as in snaps seem updated

rohanprasadofficial replied:
Let me check this

lyzhan7 replied:
I think ideally this PR should not have any changes to the snapshots, but one of the valid reasons why the snapshots might have changed in this PR is that when this other PR got introduced: https://github.com/microsoft/fluentui-react-native/pull/2242/files we switched from using windows token values to using iOS token values (which were outdated at the time), which changed a bunch of snapshots. I think some of the changes you've made are just reverting the values to what they used to be, which is what we want.

rohanprasadofficial replied:
Yes make sense.

ankraj12 commented on packages/components/Button/src/ButtonTokens.android.ts:
```diff
@@ -9,68 +9,68 @@ export const defaultButtonTokens: TokenSettings<ButtonTokens, Theme> = () =>
     },
     small: {
-      padding: globalTokens.spacing.xs - globalTokens.stroke.width10,
+      padding: globalTokens.size40 - globalTokens.stroke.width10,
```
This is specific style file for android, we should map XS to size80 in references here.

ankraj12 replied:
This also applies to references for :
`globalTokens.spacing.s` (value 12), which should now be "`globalTokens.size120`"

rohanprasadofficial replied:
I have to update the platform specific mappings, currently everything is mapped to win32/mac. Will push changes.

ankraj12 commented on packages/components/Notification/src/NotificationTokens.ios.ts:
```diff
@@ -105,8 +105,8 @@ export const defaultNotificationTokens: TokenSettings<NotificationTokens, Theme>
     minHeight: 52,
-    padding: globalTokens.spacing.m,
-    paddingVertical: globalTokens.spacing.s,
+    padding: globalTokens.size120,
```
This should map to "`globalTokens.size160`". Given there is platform styling applied already, I think it will be better to map it to right translations based on endpoint for these. For components where there is no platform specific overrides, it's safe to map them all to win32 default.

lyzhan7 replied:
I'll double check what we should do here- iOS Notification was built before the token work started so I think it might have been referencing the old win32 tokens

rohanprasadofficial replied:
I have updated it to new "ios" size ramp based on tshirt size mapping. 
m->  size160
s -> size120

lyzhan7 replied:
I double checked and for Notification specifically what you have now looks good - from this [PR](https://github.com/microsoft/fluentui-react-native/pull/2251/files) the padding should be 16 and padding vertical should be 12. The snapshots don't seem to match though, think they just need to be updated again

rohanprasadofficial replied:
Works , @lyzhan7, 
i have mapping of  padding to 16 and padding vertical to 12 now. 

Can you once verify after taking latest once i merge this PR. Thanks

ankraj12 reviewed (changes requested):
Please review mapping tokens in platform styling files.

amgleitman commented on packages/theming/theme-types/src/Theme.types.ts:
```diff
@@ -11,21 +11,31 @@ type SpacingValue = number | string;
-  xxl?: SpacingValue;
-  xxxl?: SpacingValue;
-  xxxxl?: SpacingValue;
+  sizeNone?: SpacingValue;
```
If we want to encourage users to use global tokens for sizing, we shouldn't be adding them to the `Theme` definition.

rohanprasadofficial replied:
Reverted the new addition.  
Also as mentioned in the description, we have to take the removal of spacing from theme as separate PR which should focus on entirely removal of spacing from theming.

amgleitman commented on packages/theming/theme-types/src/Theme.types.ts:
```diff
@@ -11,21 +11,31 @@ type SpacingValue = number | string;
 export interface Spacing {
   s2: string;
   s1: string;
+  m?: string;
```
This shouldn't be optional anymore. We initially did this because both the old and new T-shirt size names had `m` as a possible value.

rohanprasadofficial replied:
Reverted the new addition.  
Also as mentioned in the description, we have to take the removal of spacing from theme as separate PR which should focus on entirely removal of spacing from theming.

rohanprasadofficial commented:
### **Update :**  

Now every default , win32, macos files mapped to win32/macos ramp up. 
Android and iOS specific files maps to the  android & iOS ramp now.  
Exception : in the files [iOS Button component](https://github.com/microsoft/fluentui-react-native/blob/main/packages/components/Button/src/ButtonTokens.ios.ts#L17) , there is a place using sNudge but there is mapping in the tshirt size naming so i have mapped it to size100 , as i made sense as sNudge lies between s (size120)  and xs (size80) for mobile.

ankraj12 commented on packages/components/Checkbox/src/CheckboxTokens.android.ts:
```diff
@@ -13,16 +13,16 @@ export const defaultCheckboxTokens: TokenSettings<CheckboxTokens, Theme> = (t: T
+      spacingLabelAfter: globalTokens.size160,
       labelIsBefore: {
-        spacingLabelBefore: globalTokens.spacing.s,
+        spacingLabelBefore: globalTokens.size160,
```
should be `size120` here as it maps "`s`"?

rohanprasadofficial replied:
Correct , Thanks 😊  
Updated.

lyzhan7 reviewed (approved):
PR looks good to me, I think some snapshots are out of date after some recent commits so just make sure to update those. I checked that the tester runs fine on iOS and there aren't any obvious issues - I think it's possible that some token values are off, but thinking the bug bash next week is where we'll go through each component and double check values. About the iOS Button sNudge issue - I think what you have is fine (using size100), and for the bug bash I'll make a note to make sure we test the iOS Button

rohanprasadofficial commented:
Update : 

Updated old snapshot that of android specific files that were previously mistakenly updated to win32 ramp. Now are pointing to new correct android ramp.


Converted FABToken.ts file match Android/iOS ramp of spacing since this file was recently changes for the bring the android tokens to common file FABToken.ts file. 

Source : 
https://github.com/microsoft/fluentui-react-native/pull/2366
